### PR TITLE
feat(core): add `mkdir.mode` permission config for detached partitions

### PR DESCRIPTION
**Pull request by debanjanc01**
Added an option to pass the mkdir mode detached partitions.
A new property key has been added: **cairo.sql.detached.mkdir.mode**
The default value is the same as for other mkdir modes, 509.

This closes [#3456](https://github.com/questdb/questdb/issues/3456)

### CODE REVIEW

**Review by marregui — CHANGES_REQUESTED** [review]
well thank you very much!

now for the fun part, the review. Just a minor request, open for discussion if you are keen. I left a comment. Everything else looks amazing.

File: core/src/test/resources/server.conf [hunk 1]
```diff
@@ -120,6 +120,7 @@ cairo.sql.insert.model.pool.capacity=128
 cairo.sql.copy.buffer.size=4m
 cairo.sql.copy.model.pool.capacity=64
 cairo.commit.mode=async
+cairo.sql.detached.mkdir.mode=777
```
[the remark this answers is not in the document]
Comment (debanjanc01) [reply core/src/test/resources/server.conf]: Hey @marregui the property name you've suggested makes more sense. I've updated it across all the changes. 
I've also added the new property to the distributable `server.conf` file.

All of the review changes are on a new commit. 
Let me know if you want me to squash the 2 commits into one. 👨‍💻

**Review by marregui — APPROVED** [review]
LGTM 

thank you for doing it. We need another pair of eyes and then we will be ready to merge!

the claim form is at the bottom of https://questdb.io/community/

**Comment by marregui** [comment]
@debanjanc01 would you please be able to refresh your fork, pull master, merge master into your branch and then push on the PR again? thank you in advance.

**Comment by debanjanc01** [comment]
> @debanjanc01 would you please be able to refresh your fork, pull master, merge master into your branch and then push on the PR again? thank you in advance.

Done!

**Comment by marregui** [comment]
Hi @debanjanc01 congratulations! it has been merged!, you are a questee.

If you found the experience fascinating, perhaps you would consider contributing more :D , these contributions are not only appreciated, they make the product better, for the whole wide world. Your name will be in the release notes. Now you get to pick anything from the issues list if you would like to. You could even decide to contribute something you would really like to have, and just need that little help to get it into the codebase, officially. Join the slack channel if you have not already, your experience is valuable there https://slack.questdb.io/ , we have 2.5k questees there already.

**Comment by debanjanc01** [comment]
Hey @marregui thanks! I'm thrilled to be a questee and would love to contribute more. 

I've joined the slack channel, thank you for sharing the link. Any pointers for going about in the community?

**Comment by marregui** [comment]
community is friendly. We questees keep an eye and jump at responding when
we know the answer to some question.

Miguel Arregui
*audentes Fortuna iuvat* <http://uk.linkedin.com/in/marregui>

**Pull request merged**